src/main.cpp

```diff
@@ -369,8 +338,8 @@ int main(int argc, const char *argv[]) {
     cerr << ex.what() << std::endl;
     return -1;
   } catch (const std::runtime_error &ex) {
-    cerr << ex.what() << std::endl;
-        return -1;
+    logger->Log(LogLevel::ERROR, "Fatal runtime error: "+ string(ex.what()));
```

Comment:
The only functional change in main.cpp. Use logger to print  exceptions (and initialize logger before the big try clause to enable this)